### ISSUE TITLE
COTR-73 Problems with HandleHit and Die methods

### ISSUE THREAD

**Issue by Elijuszek** [Nekompiliuoja/Champions-of-the-Realm#57]
Actor HandleHit and Die methods should be rearranged and overridden to ensure better accessibility for derivative classes.
